app/src/main/java/org/akvo/flow/presentation/settings/PreferencePresenter.java

```diff
@@ -136,5 +149,72 @@ public void destroy() {
         saveEnableMobileData.dispose();
         saveImageSize.dispose();
         saveKeepScreenOn.dispose();
+        unSyncedTransmissionsExist.dispose();
+        clearAllData.dispose();
+        clearResponses.dispose();
+    }
+
+    public void deleteCollectedData() {
+        unSyncedTransmissionsExist.execute(new DefaultObserver<Boolean>(){
+            @Override
+            public void onError(Throwable e) {
+                Timber.e(e);
+                view.showDeleteCollectedData();
+            }
+
+            @Override
+            public void onNext(Boolean exist) {
```

Comment:
Why not a primitive here? Also maybe use the convention of prefixing with `is...` or `has...`?

Reply:
1) We are looking if unsynced transmissions exists or not, I cannot come up with a better name except, isExisting?

2) rxjava uses Boolean better use it. This how it is in all Observers